bin/start-vm

```diff
@@ -239,29 +255,40 @@ macfull="$(sed 's/../&:/g;s/:$//' <<< $mac)"
 uuid="12345678-0000-0000-0000-${mac}"
 virtOpts+=(     "-uuid $uuid" )
 
-if [ "macos" != $running_os ]; then
-    if [ "$arch" = "$(uname -m)" ]; then
```

Comment:
Keep in mind that a developer could call `start-vm` for other architectures than the architecture used to execute this script. For example, a developer could run a x86_64 Garden Linux image while being on a ARM64 machine and vice versa. This is achieved by using the `--arch` parameter of `start-vm`, so that `$arch` can be different than `$(uname -m)` in some cases.

However, by removing the implementation in this PR and replacing it with two separate if statements that are only checking if arch is set to x86_64 or aarch64, this functionality would be lost.

Reply:
Why I can understand this statement, I do not see why this shouldn't work. You can still overwrite the arch. For example (here breakpoint set to just print the output):
```
% uname -m ; bin/start-vm --arch x86_64 .build/kvm_dev-amd64-dev-local.raw
arm64
x86_64
```

Passing the `--arch` as `x86_64` on an `AARCH64` based system.

Reply:
The old implementation has set the `-cpu` parameter to `-cpu Broadwell` for x86_64 (while being on arm64) and `-cpu cortex-a72 -machine virt` for arm64 (while being on x86_64). 

As far as I can tell, this functionality isn't given anymore with this PR, no?

Reply:
You're right, this isn't given anymore because it's useless. Furthermore, this doesn't fail. Why? Let's take a deeper look at this with some reproducible commands. 

## Functionality

### Running on `x86_64` an `aarch64` Guest
First, let's give it a try if it works. We start an `aarch64` based guest on a `x86_64` host.
```
$ uname -m ; bin/start-vm --arch aarch64 test/linux_gardenlinux_kvm-dev_arm64.utm/Images/kvm_dev-arm64-dev-local.raw ; ps aux | grep qemu
x86_64
vagrant   109635 74.4  2.9 3779020 242532 pts/1  Sl+  13:43   0:14 qemu-system-aarch64 -drive if=virtio,format=raw,file=test/linux_gardenlinux_kvm-dev_arm64.utm/Images/kvm_dev-arm64-dev-local.raw -smp 2 -m 2048 -watchdog i6300esb -nodefaults -uuid 12345678-0000-0000-0000-020bb2fb3a63 -cpu max -machine virt -monitor unix:/tmp/020bb2fb3a63.monitor,server,nowait -nographic -serial mon:stdio -drive if=pflash,format=raw,unit=0,file=/usr/share/AAVMF/AAVMF_CODE.fd,readonly=on -drive if=pflash,format=raw,unit=1,file=/home/vagrant/gardenlinux/bin/020bb2fb3a63.vars -device virtio-net-pci,netdev=net0,mac=02:0b:b2:fb:3a:63 -netdev user,id=net0,hostfwd=tcp::2223-:22,hostname=garden
```
Looks pretty good. Up & running...

### Running on `aarch64` a `x86_64` Guest
Next, let's give it a try if it works. Therefore, we start a `x86_64 ` based guest on an `aarch64` host.
```
$ uname -m ; bin/start-vm --arch x86_64 .build/kvm_dev-amd64-dev-local.raw ; ps aux | grep qemu
aarch64
admin     819457  100  0.5 3605400 1461640 pts/0 Sl+  06:43   2:05 qemu-system-x86_64 -device virtio-scsi-pci,id=scsi0 -device scsi-hd,drive=drive0,bus=scsi0.0 -drive format=raw,if=none,discard=unmap,aio=native,cache.direct=on,id=drive0,file=.build/kvm_dev-amd64-dev-local.raw -smp 2 -m 2048 -watchdog i6300esb -nodefaults -device virtio-balloon -device virtio-rng-pci,rng=rng0 -object rng-random,id=rng0,filename=/dev/random -device ipmi-bmc-sim,id=bmc0 -device isa-ipmi-kcs,bmc=bmc0,ioport=0xca2 -chardev socket,path=/tmp/02:c6:37:bf:4e:c7.guest,server=on,wait=off,id=qga0 -device virtio-serial -device virtserialport,chardev=qga0,name=org.qemu.guest_agent.0 -uuid 12345678-0000-0000-0000-02c637bf4ec7 -monitor unix:/tmp/02c637bf4ec7.monitor,server,nowait -nographic -serial mon:stdio -device virtio-net-pci,netdev=net0,mac=02:c6:37:bf:4e:c7 -netdev user,id=net0,hostfwd=tcp::2223-:22,hostname=garden
```
This good, too. Up & running... Why does it work? Because `QEMU` uses the default CPU type on `x86_64` platforms.
 
### CPU Type
You're right, we do not have `-cpu cortex-a72 -machine virt` anymore in that way. But why you want to get rid of this enhancement? Based on a defined `--arch aarch64` arg, we may encounter the following cases:
 * Native Support
 * Emulation

#### Native Support
So, if `--arch aarch64` is se and we have native support, we `do not` want to emulate a specific CPU model/type and slow down for emulation. Therefore, we want to proceed with `-cpu host` in addition of an accelerator like `kvm` oder `hvf`.

#### Emulation
If we're running `--arch aarch64` in emulation on a `x86_64` system, we may `-cpu cortex-a72`. It works, but why not just set it to `-cpu max`? In this case `QEMU` will takeover for the feature probing.

Reply:
Ok, looks good to me.